astrosat_users/views/views_customers.py

```diff
@@ -64,9 +78,16 @@ class CustomerCreateView(CustomerViewMixin, generics.CreateAPIView):
     lookup_field = "id"
     lookup_url_kwarg = "customer_id"
 
-    permission_classes = [IsAuthenticated]
+    permission_classes = [IsAuthenticated, HasPendingCustomer]
     serializer_class = CustomerSerializer
 
+    def perform_create(self, serializer):
+        customer = serializer.save()
+        user = self.request.user
+        if user.pending_customer is True:
+            user.pending_customer = False
```

Comment:
I am a little confused about this, thought I understood it till I saw this line. I thought we were marking some users as having the ability to create users. But this code is checking if a user has the pending flag, if true, then set it to false and save the user. Might be better to have a telecon about this, so I do understand properly at some time.

Reply:
You are sort of right.  I am not marking a user as having an ability to create customers.  I am marking a user as being in the middle of creating a specific customer (one that is "pending").  If the `serializer.save()` line succeeds, then this user has just successfully created that customer and I can mark that user as no longer being in the middle of creating a customer.